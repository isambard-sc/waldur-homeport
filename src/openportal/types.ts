export interface ProjectInfo {
  project: string;
  shortname: string;
  allowed_destinations: string;
}

export interface UserInfo {
  user: string;
  shortname: string;
}
