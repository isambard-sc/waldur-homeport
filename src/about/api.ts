import { get } from '@waldur/core/api';

export async function getUserAgreement(agreement_type) {
  const response = await get('/user-agreements/', {
    params: { agreement_type },
  });
  return response.data[0];
}
