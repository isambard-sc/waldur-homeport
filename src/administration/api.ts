import Axios, { AxiosRequestConfig } from 'axios';

import { ENV } from '@waldur/configs/default';
import {
  deleteById,
  get,
  getAll,
  parseResultCount,
  post,
  put,
} from '@waldur/core/api';

export const getCustomersCount = () =>
  Axios.head(`${ENV.apiEndpoint}api/customers/`).then((response) =>
    parseResultCount(response),
  );

export const getProjectsCount = () =>
  Axios.head(`${ENV.apiEndpoint}api/projects/`).then((response) =>
    parseResultCount(response),
  );

export const getUsersCount = () =>
  Axios.head(`${ENV.apiEndpoint}api/users/`).then((response) =>
    parseResultCount(response),
  );

export const getCategoriesCount = () =>
  Axios.head(`${ENV.apiEndpoint}api/marketplace-categories/`).then((response) =>
    parseResultCount(response),
  );

export const getProviderOfferingsCount = (configs?: AxiosRequestConfig<any>) =>
  Axios.head(
    `${ENV.apiEndpoint}api/marketplace-provider-offerings/`,
    configs,
  ).then((response) => parseResultCount(response));

export const getResourcesCount = (configs?: AxiosRequestConfig<any>) =>
  Axios.head(`${ENV.apiEndpoint}api/marketplace-resources/`, configs).then(
    (response) => parseResultCount(response),
  );

export const getVersion = () =>
  get<{ version: string | number }>('/version/').then(
    (response) => response.data,
  );

export const getIdentityProviders = () =>
  getAll<{ provider }>('/identity-providers/');

export const getIdentityProvider = (type) =>
  get<{ client_id; logout_url }>(`/identity-providers/${type}/`).then(
    (response) => response.data,
  );

export const createIdentityProvider = (formData) =>
  post('/identity-providers/', formData);

export const updateIdentityProvider = (provider, formData) =>
  put(`/identity-providers/${provider}/`, formData);

export const deleteToken = (tokenURL: string) => Axios.delete(tokenURL);

export const createAdminAnnouncement = (formData) =>
  post('/admin-announcements/', formData);

export const updateAdminAnnouncement = (formData, uuid) =>
  put(`/admin-announcements/${uuid}/`, formData);

export const deleteAdminAnnouncement = (uuid) =>
  deleteById('/admin-announcements/', uuid);
