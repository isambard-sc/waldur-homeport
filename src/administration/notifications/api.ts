import { getList, post } from '@waldur/core/api';

import { NotificationMessageTemplate } from './types';

export const getNotificationMessagesTemplates = (params?: {}) => {
  return getList<NotificationMessageTemplate>(
    '/notification-messages-templates/',
    params,
  );
};

export const overrideNotificationTemplate = (url, content) => {
  return post(`${url}override/`, content);
};

export const enableNotification = (url) => {
  return post(`${url}enable/`);
};

export const disableNotification = (url) => {
  return post(`${url}disable/`);
};
