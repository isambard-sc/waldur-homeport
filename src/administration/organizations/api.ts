import { deleteById, post, put } from '@waldur/core/api';

export const removeOrganizationGroup = (uuid: string) =>
  deleteById('/organization-groups/', uuid);

export const createOrganizationGroup = (data) =>
  post('/organization-groups/', data);

export const updateOrganizationGroup = (uuid, data) =>
  put(`/organization-groups/${uuid}/`, data);
