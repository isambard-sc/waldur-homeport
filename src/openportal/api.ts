import { ENV } from '@waldur/configs/default';
import { post, put, sendForm } from '@waldur/core/api';

import { AllocationLimits } from './types';

export const setLimits = (id, data: AllocationLimits) =>
  post(`/openportal-allocations/${id}/set_limits/`, data);

export const updateAllocation = (id, data) =>
  put(`/openportal-allocations/${id}/`, data);

export const pullAllocation = (id: string) =>
  post(`/openportal-allocations/${id}/pull/`);

export const submitJob = (payload) =>
  sendForm('POST', `${ENV.apiEndpoint}api/openportal-jobs/`, payload);
