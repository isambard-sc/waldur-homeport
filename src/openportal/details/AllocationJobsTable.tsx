import { FunctionComponent, useMemo } from 'react';

import { translate } from '@waldur/i18n';
import { ResourceName } from '@waldur/resource/ResourceName';
import { ResourceState } from '@waldur/resource/state/ResourceState';
import { createFetcher } from '@waldur/table/api';
import Table from '@waldur/table/Table';
import { useTable } from '@waldur/table/useTable';

import { SubmitJobAction } from './SubmitJobAction';

export const AllocationJobsTable: FunctionComponent<{ resourceScope }> = ({
  resourceScope,
}) => {
  const columns = [
    {
      title: translate('Name'),
      render: ({ row }) => <ResourceName resource={row} />,
      copyField: (row) => row.name,
    },
    {
      title: translate('User'),
      render: ({ row }) => row.user_name || 'N/A',
    },
    {
      title: translate('Backend ID'),
      render: ({ row }) => row.backend_id || 'N/A',
    },
    {
      title: translate('State'),
      render: ({ row }) => <ResourceState resource={row} />,
    },
  ];
  const filter = useMemo(
    () => ({
      allocation_uuid: resourceScope.uuid,
    }),
    [resourceScope],
  );
  const tableProps = useTable({
    table: 'AllocationJobsTable',
    fetchData: createFetcher('openportal-jobs'),
    filter,
  });

  return (
    <Table
      {...tableProps}
      title={translate('Jobs')}
      columns={columns}
      verboseName={translate('jobs')}
      tableActions={<SubmitJobAction resource={resourceScope} />}
    />
  );
};
