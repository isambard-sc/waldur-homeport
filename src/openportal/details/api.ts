import { getAll } from '@waldur/core/api';

import { UserUsage } from './types';

export const getAllocationUserUsages = (params) =>
  getAll<UserUsage>('/openportal-allocation-user-usage/', { params });
