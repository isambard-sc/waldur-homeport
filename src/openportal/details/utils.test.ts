import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';

import * as marketplaceApi from '@waldur/marketplace/common/api';
import { ComponentUsage } from '@waldur/marketplace/resources/usage/types';
import * as openportalApi from '@waldur/openportal/details/api';

import componentUsages from './fixtures/component-usages.json';
import userUsages from './fixtures/user-usages.json';
import { loadCharts } from './utils';

vi.mock('@waldur/openportal/details/api');
vi.mock('@waldur/marketplace/common/api');

describe('OpenPortal allocation usage chart formatter', () => {
  beforeEach(() => {
    vi.useFakeTimers();
  });

  afterEach(() => {
    vi.useRealTimers();
  });

  beforeEach(() => {
    vi.setSystemTime(new Date(2020, 6, 1));
    vi.mocked(openportalApi).getAllocationUserUsages.mockResolvedValue(userUsages);
    vi.mocked(marketplaceApi).getComponentUsages.mockResolvedValue(
      componentUsages as ComponentUsage[],
    );
  });

  it('parses data and returns eChart option correctly', async () => {
    const charts = await loadCharts('allocationUrl', 'resourceUuid');
    expect(charts).toMatchSnapshot();
  });
});
