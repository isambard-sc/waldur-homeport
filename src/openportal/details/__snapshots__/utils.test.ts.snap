// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OpenPortal allocation usage chart formatter > parses data and returns eChart option correctly 1`] = `
[
  {
    "field": "cpu_usage",
    "name": "CPU usage",
    "options": {
      "color": [
        "#003366",
        "#006699",
        "#23c6c8",
        "#eee8d5",
        "#1c84c6",
        "green",
        "red",
        "orange",
        "yellow",
        "purple",
        "aqua",
        "pink",
      ],
      "legend": {
        "data": [
          "Usage",
          "douglas",
          "steven",
          "corey",
        ],
      },
      "series": [
        {
          "data": [
            16.67,
            33.33,
            66.67,
            33.33,
            0,
            0,
          ],
          "name": "Usage",
          "type": "line",
          "yAxisIndex": 1,
        },
        {
          "data": [
            0,
            0,
            25,
            0,
            16.67,
            0,
          ],
          "name": "douglas",
          "type": "bar",
          "yAxisIndex": 0,
        },
        {
          "data": [
            0,
            16.67,
            0,
            66.67,
            16.67,
            0,
          ],
          "name": "steven",
          "type": "bar",
          "yAxisIndex": 0,
        },
        {
          "data": [
            0,
            0,
            8.33,
            0,
            0,
            0,
          ],
          "name": "corey",
          "type": "bar",
          "yAxisIndex": 0,
        },
      ],
      "tooltip": {
        "axisPointer": {
          "crossStyle": {
            "color": "#999",
          },
          "type": "cross",
        },
        "trigger": "axis",
      },
      "xAxis": [
        {
          "axisPointer": {
            "type": "shadow",
          },
          "data": [
            "2 - 2020",
            "3 - 2020",
            "4 - 2020",
            "5 - 2020",
            "6 - 2020",
            "7 - 2020",
          ],
          "type": "category",
        },
      ],
      "yAxis": [
        {
          "axisLabel": {
            "formatter": "{value} h",
          },
          "axisLine": {
            "show": true,
          },
          "axisTick": {
            "show": true,
          },
          "name": "Usage",
          "type": "value",
        },
        {
          "axisLabel": {
            "formatter": "{value} h",
          },
          "axisLine": {
            "show": true,
          },
          "axisTick": {
            "show": true,
          },
          "name": "",
          "type": "value",
        },
      ],
    },
  },
  {
    "field": "gpu_usage",
    "name": "GPU usage",
    "options": {
      "color": [
        "#003366",
        "#006699",
        "#23c6c8",
        "#eee8d5",
        "#1c84c6",
        "green",
        "red",
        "orange",
        "yellow",
        "purple",
        "aqua",
        "pink",
      ],
      "legend": {
        "data": [
          "Usage",
          "douglas",
          "steven",
          "corey",
        ],
      },
      "series": [
        {
          "data": [
            16.67,
            33.33,
            66.67,
            33.33,
            0,
            0,
          ],
          "name": "Usage",
          "type": "line",
          "yAxisIndex": 1,
        },
        {
          "data": [
            0,
            0,
            25,
            0,
            16.67,
            0,
          ],
          "name": "douglas",
          "type": "bar",
          "yAxisIndex": 0,
        },
        {
          "data": [
            0,
            16.67,
            0,
            66.67,
            16.67,
            0,
          ],
          "name": "steven",
          "type": "bar",
          "yAxisIndex": 0,
        },
        {
          "data": [
            0,
            0,
            8.33,
            0,
            0,
            0,
          ],
          "name": "corey",
          "type": "bar",
          "yAxisIndex": 0,
        },
      ],
      "tooltip": {
        "axisPointer": {
          "crossStyle": {
            "color": "#999",
          },
          "type": "cross",
        },
        "trigger": "axis",
      },
      "xAxis": [
        {
          "axisPointer": {
            "type": "shadow",
          },
          "data": [
            "2 - 2020",
            "3 - 2020",
            "4 - 2020",
            "5 - 2020",
            "6 - 2020",
            "7 - 2020",
          ],
          "type": "category",
        },
      ],
      "yAxis": [
        {
          "axisLabel": {
            "formatter": "{value} h",
          },
          "axisLine": {
            "show": true,
          },
          "axisTick": {
            "show": true,
          },
          "name": "Usage",
          "type": "value",
        },
        {
          "axisLabel": {
            "formatter": "{value} h",
          },
          "axisLine": {
            "show": true,
          },
          "axisTick": {
            "show": true,
          },
          "name": "",
          "type": "value",
        },
      ],
    },
  },
  {
    "field": "ram_usage",
    "name": "RAM usage",
    "options": {
      "color": [
        "#003366",
        "#006699",
        "#23c6c8",
        "#eee8d5",
        "#1c84c6",
        "green",
        "red",
        "orange",
        "yellow",
        "purple",
        "aqua",
        "pink",
      ],
      "legend": {
        "data": [
          "Usage",
          "douglas",
          "steven",
          "corey",
        ],
      },
      "series": [
        {
          "data": [
            0.02,
            0.03,
            0.07,
            0.03,
            0,
            0,
          ],
          "name": "Usage",
          "type": "line",
          "yAxisIndex": 1,
        },
        {
          "data": [
            0,
            0,
            0.02,
            0,
            0.02,
            0,
          ],
          "name": "douglas",
          "type": "bar",
          "yAxisIndex": 0,
        },
        {
          "data": [
            0,
            0.02,
            0,
            0.07,
            0.02,
            0,
          ],
          "name": "steven",
          "type": "bar",
          "yAxisIndex": 0,
        },
        {
          "data": [
            0,
            0,
            0.01,
            0,
            0,
            0,
          ],
          "name": "corey",
          "type": "bar",
          "yAxisIndex": 0,
        },
      ],
      "tooltip": {
        "axisPointer": {
          "crossStyle": {
            "color": "#999",
          },
          "type": "cross",
        },
        "trigger": "axis",
      },
      "xAxis": [
        {
          "axisPointer": {
            "type": "shadow",
          },
          "data": [
            "2 - 2020",
            "3 - 2020",
            "4 - 2020",
            "5 - 2020",
            "6 - 2020",
            "7 - 2020",
          ],
          "type": "category",
        },
      ],
      "yAxis": [
        {
          "axisLabel": {
            "formatter": "{value} GB-hours",
          },
          "axisLine": {
            "show": true,
          },
          "axisTick": {
            "show": true,
          },
          "name": "Usage",
          "type": "value",
        },
        {
          "axisLabel": {
            "formatter": "{value} GB-hours",
          },
          "axisLine": {
            "show": true,
          },
          "axisTick": {
            "show": true,
          },
          "name": "",
          "type": "value",
        },
      ],
    },
  },
]
`;
